src/pool/srv_pool.c

```diff
@@ -5064,30 +5067,15 @@ __ds_pool_mark_upgrade_completed(uuid_t pool_uuid, struct pool_svc *svc, int rc)
 	return rc1;
 }
 
-/* check and upgrade the object layout if needed. */
+/* Schedule durable format check on all targets. */
 static int
-pool_check_upgrade_object_layout(struct rdb_tx *tx, struct pool_svc *svc,
-				 bool *scheduled_layout_upgrade)
+pool_upgrade_schedule(struct pool_svc *svc)
 {
 	daos_epoch_t	upgrade_eph = d_hlc_get();
-	d_iov_t		value;
-	uint32_t	current_layout_ver = 0;
 	int		rc = 0;
 
-	d_iov_set(&value, &current_layout_ver, sizeof(current_layout_ver));
-	rc = rdb_tx_lookup(tx, &svc->ps_root, &ds_pool_prop_obj_version, &value);
-	if (rc && rc != -DER_NONEXIST)
-		return rc;
-	else if (rc == -DER_NONEXIST)
-		current_layout_ver = 0;
-
-	if (current_layout_ver < DS_POOL_OBJ_VERSION) {
```

Comment:
why remove this check?

Reply:
pool_upgrade_schedule() is moved to ds_pool_upgrade_if_needed().  So all upgrade will go through the "rebuild/upgrade" process if needed, to upgrade vos_container on every vos target. @wangshilong  Could you please check this changes as well?

Reply:
Looks ok to me.